bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller_test.go

```diff
@@ -1741,55 +1766,83 @@ func newWorkerMachinePool(cluster *clusterv1.Cluster) *expv1.MachinePool {
 }
 
 // newKubeadmConfig return a CABPK KubeadmConfig object; if machine is not nil, the KubeadmConfig is linked to the machine as well.
-func newKubeadmConfig(machine *clusterv1.Machine, name, namespace string) *bootstrapv1.KubeadmConfig {
+func newKubeadmConfig(namespace, name string) *bootstrapv1.KubeadmConfig {
 	return bootstrapbuilder.KubeadmConfig(namespace, name).
-		WithMachine(machine).
 		Build()
 }
 
-func newWorkerJoinKubeadmConfig(machine *clusterv1.Machine) *bootstrapv1.KubeadmConfig {
-	return bootstrapbuilder.KubeadmConfig(machine.Namespace, "worker-join-cfg").
+func newWorkerJoinKubeadmConfig(namespace string) *bootstrapv1.KubeadmConfig {
```

Comment:
This seems a little bit weird given that it accepts only a namespace and not name.
Should we standardise?